packages/augur-core/source/contracts/matic/AugurPredicate.sol

```diff
@@ -4,74 +4,86 @@ pragma experimental ABIEncoderV2;
 
     bytes32 constant SHARE_TOKEN_BALANCE_CHANGED_EVENT_SIG = 0x350ea32dc29530b9557420816d743c436f8397086f98c96292138edd69e01cb3;
-    uint256 MAX_LOGS = 10;
+    uint256 MAX_LOGS = 100;
 
     Registry public registry;
+    IWithdrawManager public withdrawManager;
 
     IAugur public augur;
-    ShareToken public shareToken;
+    IShareToken public shareToken;
     IZeroXTrade public zeroXTrade;
+    Cash public cash;
+    OICash public oICash;
+
+    mapping(address => bool) claimedTradingProceeds;
 
     struct ExitData {
-        address shareToken;
-        address cash;
+        IShareToken shareToken;
+        Cash cash;
+        uint256 exitPriority;
+        bool exitInitiated;
+        IMarket[] marketsList;
+        mapping(address => bool) markets;
     }
 
     mapping(uint256 => ExitData) public lookupExit;
 
-    function initialize(IAugur _augur, IAugurTrading _augurTrading) public beforeInitialized {
+    function initialize(IAugur _augur, IAugurTrading _augurTrading, OICash _oICash) public beforeInitialized {
         endInitialization();
         augur = _augur;
-        // This ShareToken is the real slim shady (from mainnet augur)
-        shareToken = ShareToken(_augur.lookup("ShareToken"));
+        shareToken = IShareToken(_augur.lookup("ShareToken"));
         zeroXTrade = IZeroXTrade(_augurTrading.lookup("ZeroXTrade"));
+        cash = Cash(_augur.lookup("Cash"));
+        oICash = _oICash;
     }
 
-    function setRegistry(address _registry) public /* @todo make part of initialize() */ {
+    function initializeForMatic(address _registry, address _withdrawManager) public /* @todo make part of initialize() */ {
         registry = Registry(_registry);
+        withdrawManager = IWithdrawManager(_withdrawManager);
     }
 
     /**
      * @notice Call initializeForExit to instantiate new shareToken and Cash contracts to replay history from Matic
      * @dev new ShareToken() / new Cash() causes the bytecode of this contract to be too large, working around that limitation for now,
-        however, the intention is to deploy a new ShareToken and Cash contract per exit
-     * @param market Market address to start the exit for
+        however, the intention is to deploy a new ShareToken and Cash contract per exit - todo: use proxies for that
      */
-    // function initializeForExit(address market) external returns(uint256 exitId) {
-    function initializeForExit(address market, address shareToken_, address cash_) external returns(uint256 exitId) {
-        (address _rootMarket, uint256 _numOutcomes, uint256 _numTicks) = registry.childToRootMarket(market);
-        require(
-            _rootMarket != address(0x0),
-            "AugurPredicate:initializeForExit: Market is not mapped"
-        );
-        exitId = getExitId(market, msg.sender);
+    function initializeForExit(IShareToken _shareToken, Cash _cash) external returns(uint256 exitId) {
```

Comment:
These are the actual shareToken and cahs contracts rather than the predicate ones which are stored in this contract? Perhaps naming each prefixed with what set they belong to would help (e.g "predicateCash", "AugurCash", etc)

Reply:
Done.